### DISCUSSION TITLE
[HELP]: StrongSwan to Forcepoint IPsec

### DISCUSSION THREAD

**Discussion by jhay06** [strongswan/strongswan#1060] [Q&A]
Hi ,
  Would like to request an assistance in setting up strongswan to connect to Force point IP Sec VPN 

Currently for forcepoint to connect to our company network this is the process 

1. We create a certificate request in basic mode as seen below
![image](https://user-images.githubusercontent.com/5191437/169997064-71eb0513-2508-4083-9aed-7fcd5c602f72.png)
2. We enter username and passphrase 
![image](https://user-images.githubusercontent.com/5191437/169997232-1878ac6e-e6d6-4f76-8739-227662804967.png)
3. then we will submit our csr file to our administrator and then they will send us signed crt file 
4. then we will import crt file 
5. then click connect to new gateway 
6. enter hostname and set authenticaion to the email address we set from the certificate
![image](https://user-images.githubusercontent.com/5191437/169997695-bf6dc345-d8bc-4c37-a1a3-c1a1a796ad21.png)
7. Then after that we can connect to our network , 

by deeper investigation i can find the private key file used by the certificate and the public .ca file 

using charon-cmd i can connect to network using below command 

sudo charon-cmd --host {peers[peer_id]} --identity {identity} \
--ike-proposal {ike_algo} --esp-proposal {esp_algo} --xauth-username {identity} \
 --cert {cert_file} --rsa {priv_file} --profile {profile} \
--cert {pub_files[0]} --cert {pub_files[1]}  {remote_ts.strip()} \
--debug 3

but the IP assigned is wrong ,  the IP given is 172.XXX.XXX.XXX/32 but i expect 172.XXX.0.0/20 

now i try to use the strongswan using ipsec.conf
but i can't manage to connect to network 
sorry for the bad english 

hope that someone enlighten me.

Thanks

**Comment by jhay06** [comment]
by the way , by checking the ike version is 1 , the ike algo is 3des-sha1 and the esp algo is 3des-sha1 also

Thnaks

**Comment by tobiasbrunner** [comment]
What problem do you have exactly? Seems straight forward to translate the charon-cmd options to settings in ipsec.conf/ipsec.secrets (or even better swanctl.conf). Please consult #196.

Note that you are using deprecated protocols with deprecated algorithms.

**Comment by jhay06** [comment]
yes i know that ikev1 is deprecated but the company is still using this protocol, 

the problem on the charon is that i can connect to the network using this command 
sudo charon-cmd --host domain.com --identity me@domain.com
--ike-proposal 3des-sha1-modp1024  --esp-proposal 3des-sha1-modp1024 --xauth-username me@domain.com
--cert  mysigned.crt --rsa myprivatekey.prv --profile ikev1-pub
--cert  companypub1.ca --cert companypub2.ca  -remote-ts 10.10.10.1/32 --remote-ts 10.10.0.0/20
--debug 3


but once i connected on the vpn , charon-cmd is giving me virtual ip with subnet 32 which is wrong ,
i expected subnet 20 , so i tried to add --local-ts 10.20.0.0/20 but this will cause to connect and immediately disconnect charon-cmd 

our company is giving us static virtual ip with subnet 20 so my ip is fix for example 10.20.1.8 

and checking on the logs since virtual ip has wrong subnet , i can see that querying policy from 10.20.XX.XX/32 == 10.10.10.XX/32 is failing since that is not the expected ip of the company 


Question i have now , 

1. does charon-cmd supports to set virtual ip in static ? 
2. how can i translate it to swanctl maybe it will work on swanctl 
3. how can i translate it to ipsec.conf maybe it will work on ipsec command ? 


sorry , i can't attach any because its the company policy here in my country . 

Thanks

**Reply by tobiasbrunner** [reply]
Using "static" virtual IPs is a simple as assigning that IP to any local interface and setting the appropriate local traffic selector (see [the docs](https://docs.strongswan.org/docs/5.9/features/vip.html)).

The translation to either config file format is straight forward, just configure the corresponding settings (see [here](https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html) for swanctl.conf).
